clients/venice-client/src/main/java/com/linkedin/venice/fastclient/RetriableAvroGenericStoreClient.java

```diff
@@ -175,6 +175,30 @@ protected CompletableFuture<V> get(GetRequestContext requestContext, K key) thro
    */
   protected CompletableFuture<Map<K, V>> batchGet(BatchGetRequestContext<K, V> requestContext, Set<K> keys)
       throws VeniceClientException {
+    CompletableFuture<Map<K, V>> responseFuture = new CompletableFuture<>();
+    CompletableFuture<VeniceResponseMap<K, V>> streamingResponseFuture = streamingBatchGet(requestContext, keys);
+    streamingResponseFuture.whenComplete((response, throwable) -> {
+      if (throwable != null) {
+        responseFuture.completeExceptionally(throwable);
+      } else if (!response.isFullResponse()) {
+        if (requestContext.getPartialResponseException().isPresent()) {
```

Comment:
Since the error message is the same and `VeniceClientException` supports passing in a null for exception.  Why don't we just replace this `if else` block to a single:
```
responseFuture.completeExceptionally(
              new VeniceClientException(
                  "Response was not complete",
                  requestContext.getPartialResponseException().orElse(null)));
```

Reply:
Modified this.